android/app/src/main/java/au/com/robin/sms/service/SubscriberService.kt

```diff
@@ -130,7 +134,8 @@ class SubscriberService : Service() {
             }
         val alarmManager = getSystemService(ALARM_SERVICE) as AlarmManager
         // Open connection
-        WsConnection(alarmManager).start()
+        wsConnection = WsConnection(::onMessageReceived, alarmManager)
```

Comment:
I assume :: means self or something?

Reply:
I believe it's not as what I understand about `self` is that it refers to the the instance within a class (correct me if I am wrong). But, in this case, it is just a reference to a function so that we can pass around or store in in a variables. Hope that is clear enough.

Reply:
I also think you can just use an anonymous function here, no need to use a class method (onMessageReceived)